src/NUnitFramework/framework/Internal/Filters/CompositeFilter.cs

```diff
@@ -64,6 +64,24 @@ public void Add(ITestFilter filter)
         public IList<ITestFilter> Filters { get; private set; }
 
         /// <summary>
+        /// Checks whether the CompositeFilter is matched by a test.
+        /// </summary>
+        /// <param name="test">The test to be matched</param>
+        public abstract override bool Pass(ITest test);
```

Comment:
Except it's almost certain that all the classes we will ever have

In src/NUnitFramework/framework/Internal/Filters/CompositeFilter.cs
<https://github.com/nunit/nunit/pull/1400#discussion_r59176879>:

> @@ -64,6 +64,24 @@ public void Add(ITestFilter filter)
>          public IList<ITestFilter> Filters { get; private set; }
>
>          /// <summary>
> +        /// Checks whether the CompositeFilter is matched by a test.
> +        /// </summary>
> +        /// <param name="test">The test to be matched</param>
> +        public abstract override bool Pass(ITest test);

I think that's a matter of preference. I personally prefer it even in
purely internal code to force implementers to override methods that doesn't
make sense how they are implemented in the base class.

But I can drop that from the commit, that's not a big deal [image: :smile:].

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/nunit/nunit/pull/1400/files/9ca0604beef21f693409647b3389283c4333caf8#r59176879>